### PR TITLE
support specreduce 1.3

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request makes necessary changes to support the (upcoming) 1.3 release of specreduce.

This combines (and replaces) #1695 and #1847.

This will
* add support for `specutils.Spectrum1D` objects as input/output to specreduce from https://github.com/astropy/specreduce/pull/144. This will also enable us to support keeping the spectral-axis/WCS throughout the spectral extraction process (although that is temporarily disabled here until the proper plotting support is implemented upstream).
* add support for `model_type` from the `FitTrace` implementation in https://github.com/astropy/specreduce/pull/128.

Todo:
- [x] fix units assertion error in `test_parser`
- [x] fix 1d background subtraction mapping to pixels
- [x] add validation rules for number of bins
- [x] remove temporary specreduce-dev testing

Once specreduce 1.3 is released, this PR should pin that new release and should then be tested (although it does currently seem to work as expected with the current state of the PR branch).

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
I think you also need https://github.com/astropy/specreduce/pull/151

**Comment by kecnry** [comment]
that isn't merged yet, so I don't know if it'll make 1.3 or not.  What changes would that need on the jdaviz-side?

**Comment by pllim** [comment]
See https://github.com/spacetelescope/jdaviz/issues/1877#issuecomment-1332604985

**Comment by pllim** [comment]
To be sure, you can temporarily add back dev specreduce job here.

**Comment by pllim** [comment]
I don't see any testing against specreduce dev, does this mean specreduce 1.3 is released?

**Comment by kecnry** [comment]
Yes, specreduce 1.3 has been released and pinned as part of this PR.  The temporary testing against dev has therefore been removed.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py [hunk 1]
```diff
@@ -792,12 +807,15 @@ def export_bg_spectrum(self, add_data=False, **kwargs):
...
             defined in the plugin.
         """
-        bg = self.export_bg(**kwargs)
-        spec = bg.bkg_spectrum()
+        spec = self.export_bg(**kwargs).bkg_spectrum()
 
         if add_data:
             self.bg_spec_add_results.add_results_from_plugin(spec, replace=False)
 
+        # TEMPORARY: override spectral axis to be in pixels until properly supporting plotting
```
Comment (pllim) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:815]: What issue will fix this? Is this future work?

**Review by pllim — APPROVED** [review]
I say we get this in now to get the CI green and worry about little bugs later.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:815]: This is the general effort to support wavelengths (waiting for glue/bqplot work in order to sync the axes correctly).  So yes, this is intentionally added to keep the current behavior.

**Comment by kecnry** [comment]
Just to clarify - CI on main/other PRs should not be broken as we pin a minor version of specreduce.  This PR just aims to make use of the new functionality and cleaner API in specreduce 1.3 and keep us up-to-date.

**Comment by kecnry** [comment]
Found one other place that needs fixing/updating: if using Horne extraction on the original data (not "From Plugin"), then specreduce fails with the following error: `image NDData object lacks uncertainty`

To reproduce:

```
from jdaviz import Specviz2d
specviz2d = Specviz2d()

from astroquery.mast import Observations 
fn = "jw01538-o160_s00004_nirspec_f170lp-g235h-s1600a1-sub2048_s2d.fits"
uri = f"mast:JWST/product/{fn}"
result = Observations.download_file(uri)

specviz2d.load_data(fn)
specviz2d.show()

spex = specviz2d.plugins['Spectral Extraction']
spex.open_in_tray()
spex.ext_type = 'Horne'
spex.ext_dataset = 'Spectrum 2D'
spex.export_extract_spectrum()
```

**Review by duytnguyendtn — APPROVED** [review]
I installed this PR and successfully grabbed specreduce=1.3.0. Running pytest `jdaviz` resulted in no errors. 

My only thought is whether you wanted to add a new test to cover the case you found in https://github.com/spacetelescope/jdaviz/pull/1889#issuecomment-1343100000?

**Comment by kecnry** [comment]
> Was the choice to leave out Chebyshev models intentional? I'll hold off on the approval in case the answer is no because I think it would be good to have parity in the available trace models.

I honestly can't remember, but it's trivial to expose (now or later).  @PatrickOgle @camipacifici - do we want all supported models in jdaviz or just some subset.  I think specreduce's `FitTrace` currently supports Spline, Polynomial, Legendre, and Chebyshev from https://github.com/astropy/specreduce/pull/128.

> I can get the UI to stop responding for several seconds if I change trace parameters too quickly. Can you replicate that? I don't see any unexpected warnings, and I'm guessing part of the issue is lag from FitTrace's calculation of fits in certain scenarios.

Yes, with binning disabled, the interactivity can start to feel laggy.  It was largely for this reason that I have binning enabled by default, even though it isn't enabled by default in `specreduce`.  Seconds seems extreme, though, and longer than I've experienced with the sample data I've used for testing - if that is the case, perhaps we should force binning or implement a spinner (not sure if a spinner would be in scope here or a follow-up effort though).

**Comment by kecnry** [comment]
@ojustino - I added Chebyshev support and also a warning in the plugin for possible slow behavior.  I think I'd rather defer a possible spinner until we see how it performs in the wild.

**Comment by kecnry** [comment]
RTD was passing before and no changes were made to docs since, so I think that is safe to ignore for now.

**Comment by pllim** [comment]
I restarted RTD, just to see if it is transient.

**Pull request merged**